### PR TITLE
[16.0][IMP] account_statement_base: add form view for account_bank_statement

### PR DESCRIPTION
**Pull request by sergiobstoj**
Added a form view for account_bank_statement model

### CODE REVIEW

File: account_statement_base/views/account_bank_statement.xml [hunk 1]
```diff
@@ -11,4 +11,144 @@
...
+                            <field name="is_complete" on_change="1" readonly="1" />
+                        </group>
+                        <group>
+                            <field name="is_valid" on_change="1" readonly="1" />
+                        </group>
+                    </group>
+                    <group>
+                        <field name="problem_description" readonly="1" />
+                    </group>
+                </sheet>
```
Comment (aritzolea) [inline account_statement_base/views/account_bank_statement.xml:54]: Faltaría un último grupo con `attachment_ids`

```
<group>
    <field name="attachment_ids"/>
</group>
```

File: account_statement_base/views/account_bank_statement.xml [hunk 2]
```diff
@@ -11,4 +11,144 @@
...
+                            <field name="reference" />
+                            <field name="first_line_index" readonly="1" />
+                            <field name="balance_end" on_change="1" readonly="1" />
+                            <field name="company_id" readonly="1" />
+                            <field name="journal_id" on_change="1" readonly="1" />
+                        </group>
+                    </group>
+                    <group>
+                        <field name="line_ids" on_change="1" required="1">
+                            <tree editable="top" multi_edit="1">
```
Comment (aritzolea) [inline account_statement_base/views/account_bank_statement.xml]: No definas el tree, solamente deja el elemento `<field name="line_ids" .....>`, así utiliza la vista para las líneas de define este mismo módulo

**Review by aritzolea — APPROVED** [review]

**Review by MayteGLC — APPROVED** [review]

File: account_statement_base/views/account_bank_statement.xml [hunk 3]
```diff
@@ -11,4 +11,49 @@
...
+                        </group>
+                        <group>
+                            <field name="is_valid" on_change="1" readonly="1" />
+                        </group>
+                    </group>
+                    <group>
+                        <field name="problem_description" readonly="1" />
+                    </group>
+                    <group>
+                        <field name="attachment_ids"/>
```
Comment (pedrobaeza) [inline account_statement_base/views/account_bank_statement.xml]: Por qué no utilizar el chatter normal?

Comment (sergiobstoj) [reply account_statement_base/views/account_bank_statement.xml]: hemos creado otro PR en el que hemos migrado el modulo account_bank_statement_chatter y añade el chatter.
https://github.com/OCA/account-financial-tools/pull/1869

Comment (pedrobaeza) [reply account_statement_base/views/account_bank_statement.xml]: Oh, in this repo we need to talk in English, sorry. Isn't the statement inheriting from `mail.thread`. If so, better to include it here.

Comment (sergiobstoj) [reply account_statement_base/views/account_bank_statement.xml]: In this module, `account.bank.statement` is not inheriting from `mail.thread`, neither from `account` module. The PR mentioned above is the migration of  account_bank_statement_chatter module. We think is better to put that dependencie on that module.

**Comment by sergiobstoj** [comment]
@pedrobaeza is this OK for merge ?

**Comment by sergio-teruel** [comment]
Hi @SergioBustamanteFL 

I think that this view show fields not needed.. the view is not clear...

I have opened a PR (I didn't know of this) with a more simple view like this.

With correct display warning, smart button to access to lines directly...

image

Also... if you add a line you have an error due to Odoo error. I will open a PR to Odoo... 

If you want to improve your eyesight, perfect, if not we can continue in my PR.

**Comment by sergio-teruel** [comment]
PR to odoo to fix the creation of statement lines inside the statement form view https://github.com/odoo/odoo/pull/167224

**Comment by sergiobstoj** [comment]
@sergio-teruel I like your approach. Let's continue in your PR. Thank you!

**Pull request closed without merging**